src/components/HURUmap/IndicatorTitle/Download.js

```diff
@@ -0,0 +1,187 @@
+
+  const [view, setView] = useState(null);
+  useEffect(() => {
+    setView(viewProp);
+  }, [viewProp]);
+
+  const handleImageDownload = async (e, type) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const imgType = type.toLowerCase();
+
+    const url = await view.toImageURL(imgType);
+    const link = document.createElement("a");
+    link.download = `${title}.${imgType}`;
+    link.href = url;
+    document.body.appendChild(link);
+    link.click();
+    document.body.removeChild(link);
+  };
+
+  const handleDataDownload = async (e, type) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const fileType = type.toLowerCase();
+
+    const data = view.data("table");
+    const fileName = `${title}.${fileType}`;
+    let href;
+
+    if (type === "json") {
+      href = `data:text/json;charset=utf-8,${encodeURIComponent(
+        JSON.stringify(data)
+      )}`;
+    } else if (type === "csv") {
+      href = `data:text/csv;charset=utf-8,${Papa.unparse(data)}`;
+    } else {
+      const table = XLSX.utils.json_to_sheet(data);
+      const wb = XLSX.utils.book_new(); // make Workbook of Excel
+      // add Worksheet to Workbook
+      XLSX.utils.book_append_sheet(wb, table, title);
+      // export Excel file
+      XLSX.writeFile(wb, fileName);
+      return;
+    }
+
+    const link = document.createElement("a");
+    link.href = href;
+    link.download = fileName;
+    document.body.appendChild(link);
+    link.click();
+    link.remove();
+  };
+
+  return (
+    <Grid container className={classes.root}>
+      <Grid item container xs={12} className={classes.row}>
+        {["Percentage", "Value"].map((v) => (
+          <Grid item xs={6} index={v} className={classes.button}>
+            <ButtonBase
+              className={clsx(classes.text)}
+              onClick={() => setChartValue(v)}
+            >
+              {v}
+            </ButtonBase>
+          </Grid>
+        ))}
+      </Grid>
+      <Grid item xs={12} className={clsx(classes.row, classes.header)}>
+        <Typography className={classes.text}>Download chart as:</Typography>
+      </Grid>
+      <Grid item container className={classes.row}>
+        {["PNG", "SVG"].map((p) => (
+          <Grid item xs={6} index={p} className={classes.button}>
+            <ButtonBase
+              className={classes.text}
+              onClick={(e) => handleImageDownload(e, p)}
+            >
+              {p}
+            </ButtonBase>
+          </Grid>
+        ))}
+      </Grid>
+      <Grid item xs={12} className={clsx(classes.row, classes.layout)}>
+        <Typography className={classes.text}>Layout option:</Typography>
+      </Grid>
+      <Grid item container className={classes.row}>
+        {["Layout1", "Layout2"].map((p) => (
+          <Grid item xs={6} index={p} className={classes.button}>
+            <ButtonBase
+              className={classes.text}
+              onClick={() => setChartValue(p)}
```

Comment:
Doesn't work or I don't know what these Layout options are all about.

Reply:
@kilemensi pierro, just explained it to me yesterday. I will have a story for it

Reply:
Added a story on [PT]([url](https://www.pivotaltracker.com/story/show/179750888)) for this